src/main/scala/io/github/pauljamescleary/petstore/model/OrderStatus.scala

```diff
@@ -1,20 +1,66 @@
 package io.github.pauljamescleary.petstore.model
 
 sealed trait OrderStatus extends Product with Serializable
-case object Approved extends OrderStatus
-case object Delivered extends OrderStatus
-case object Placed extends OrderStatus
 
+/*
+ * Putting case object in the object with the same name
+ * as you sealed trait is a common practice that allows
+ * - to qualify named (better discoverability for other dev)
+ *   ex: OrderStatus.Approved is self-telling, don't need to
+ *   wonder if it is not Status.Approved)
+ * - to use libraries which needs that convention
+ *   (ex: sealerate is a neat little tool that I used frequently
+ *   when I need to iterate on all cases of the ADT)
+ * - it is widelly use in scalaz/monix/shapeless/etc to better
+ *   managed implicits scoping
+ */
 object OrderStatus {
-  def apply(name: String): OrderStatus = name match {
-    case "Approved" => Approved
-    case "Delivered" => Delivered
-    case "Placed" => Placed
-  }
 
+  case object Approved extends OrderStatus
+  /**
+   * to be known: case object into other object are not final from the point
+   * of view of the JVM: javap OrderedStatus$Delivered =>
+   *   public class io.github.pauljamescleary.petstore.model.OrderStatus$Delivered$ ...
+   * So you may want to use "final case object ...", that can
+   * give the jvm better insight on optimisation.
+   */
+  case object Delivered extends OrderStatus
+  //given as example: javap OrderedStatus$Placed
+  //  public final class io.github.pauljamescleary.petstore.model.OrderStatus$Placed$
+  final case object Placed extends OrderStatus
+
+  //here, you can call
+  def allValues = ca.mrvisser.sealerate.values[OrderStatus]
```

Comment:
ahh interesting

Reply:
I didn't go here just yet.  I tinkered some with Enumeratum as well.  Moving the values inside the companion object didn't play well with circe (at least when I tried it).

I do plan on revisiting the models however, and will address this then